### ISSUE TITLE
Solid.BySweep2Rails Causes Dynamo Crash

### ISSUE THREAD

**Issue by mzjensen** [DynamoDS/Dynamo#10649]
If this issue is with Dynamo for Revit, please post your issue on the [Dynamo for Revit Issues page](https://github.com/DynamoDS/DynamoRevit/issues).

If this issue is **not** a bug report or improvement request, please check the [Dynamo forum](https://forum.dynamobim.com/), and start a thread there to discuss your issue.

## Dynamo version

2.4.1.6394

## Operating system

Windows 10

## What did you do? 

I am trying to sweep a non-circular closed polycurve along two rails with Solid.BySweep2Rails.

## What did you expect to see?

I expected to see the polycurve sweep along the rails like it does with a circular profile.

![Circular profile](https://user-images.githubusercontent.com/24361762/81828252-d420b380-94f6-11ea-9b17-99d0d05af83d.gif)

## What did you see instead?

Dynamo crashes when using a polycurve as the profile.

![Rectangular profile](https://user-images.githubusercontent.com/24361762/81827373-ef3ef380-94f5-11ea-987a-021217af1a5b.gif)

**Comment by mjkkirschner** [comment]
Hi @mzjensen - is there a stack trace that appears or does Dynamo just exist to desktop?

**Comment by mzjensen** [comment]
@mjkkirschner - in Sandbox, there are no dialogs and it just goes straight to desktop. In Civil 3D (where I am ultimately trying to use this), it gives an unhandled exception fatal error.

**Comment by mjkkirschner** [comment]
tracked with: https://jira.autodesk.com/browse/DYN-2756

**Comment by mmisol** [comment]
Hi @mzjensen . First of all, thanks for taking the time to reach out to us and provide a clear repro of the issue.

We could identify this as a special case for this function in LibG, our geometry library. We were able to prevent the crash by handling that special case appropriately. A LibG update was merged into our repo with this fix in https://github.com/DynamoDS/Dynamo/pull/10734.

Unfortunately, we don't currently support this operation with a polycurve profile. Supporting that is a longer stretch goal, because we need to first convert it into a curve supported by the underlying library we are using (ASM), and we don't have that implemented yet.
